test/backend/random_uniform.in.cpp

```diff
@@ -0,0 +1,313 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/all_close.hpp"
+#include "util/all_close_f.hpp"
+#include "util/known_element_types.hpp"
+#include "util/ndarray.hpp"
+#include "util/test_control.hpp"
+#include "util/test_tools.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+
+NGRAPH_TEST(${BACKEND_NAME}, random_uniform_all_static_seed_unused)
+{
+    auto min_val = make_shared<op::Constant>(element::f32, Shape{}, std::vector<float>{63.0f});
+    auto max_val = make_shared<op::Constant>(element::f32, Shape{}, std::vector<float>{120.0f});
+    auto result_shape =
+        make_shared<op::Constant>(element::i64, Shape{3}, std::vector<float>{50, 200, 100});
+    auto use_fixed_seed =
+        make_shared<op::Constant>(element::boolean, Shape{}, std::vector<char>{0});
+    size_t fixed_seed = 9999;
+
+    auto ru =
+        make_shared<op::RandomUniform>(min_val, max_val, result_shape, use_fixed_seed, fixed_seed);
+
+    auto f = make_shared<Function>(NodeVector{ru}, ParameterVector{});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+
+    auto ex = backend->compile(f);
+
+    auto t_r = backend->create_tensor(element::f32, Shape{50, 200, 100});
+
+    ex->call_with_validate({t_r}, {});
+    auto results_0 = read_vector<float>(t_r);
+    ASSERT_TRUE(
+        std::all_of(results_0.begin(), results_0.end(), [](float x) { return (x >= 63.0f); }));
+    ASSERT_TRUE(
+        std::all_of(results_0.begin(), results_0.end(), [](float x) { return (x <= 120.0f); }));
+
+    ex->call_with_validate({t_r}, {});
+    auto results_1 = read_vector<float>(t_r);
+    ASSERT_TRUE(
+        std::all_of(results_1.begin(), results_1.end(), [](float x) { return (x >= 63.0f); }));
+    ASSERT_TRUE(
+        std::all_of(results_1.begin(), results_1.end(), [](float x) { return (x <= 120.0f); }));
+
+    ASSERT_FALSE(test::all_close_f(results_0, results_1))
+        << "Two separate vectors of 1000000 random numbers matched exactly. This is technically "
```

Comment:
I think I'll just delete this message, or rephrase it to something like "random vectors matched even though use_fixed_seed was set". It's based on the theoretical assumption that the RNG might, maybe, who knows, be close enough to perfect that it might, maybe, who knows, occasionally generate successive identical strings of length 1000000 which we might, maybe, who knows, happen to be getting here, but gosh golly is it ever unlikely.